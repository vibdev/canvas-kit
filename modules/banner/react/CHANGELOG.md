# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.25](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.24...@workday/canvas-kit-react-banner@0.1.25) (2019-06-20)


### Bug Fixes

* broad stroke accessibility fixes ([9803412](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/commits/9803412))





## [0.1.24](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.23...@workday/canvas-kit-react-banner@0.1.24) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.23](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.22...@workday/canvas-kit-react-banner@0.1.23) (2019-06-18)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.22](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.21...@workday/canvas-kit-react-banner@0.1.22) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.21](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.20...@workday/canvas-kit-react-banner@0.1.21) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.20](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.19...@workday/canvas-kit-react-banner@0.1.20) (2019-06-14)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.19](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.18...@workday/canvas-kit-react-banner@0.1.19) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.18](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.17...@workday/canvas-kit-react-banner@0.1.18) (2019-06-13)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.17](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.16...@workday/canvas-kit-react-banner@0.1.17) (2019-06-10)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.16](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.15...@workday/canvas-kit-react-banner@0.1.16) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.15](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.14...@workday/canvas-kit-react-banner@0.1.15) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.14](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.13...@workday/canvas-kit-react-banner@0.1.14) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.13](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.12...@workday/canvas-kit-react-banner@0.1.13) (2019-05-31)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.12](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.11...@workday/canvas-kit-react-banner@0.1.12) (2019-05-24)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.11](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.10...@workday/canvas-kit-react-banner@0.1.11) (2019-05-22)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.10](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.9...@workday/canvas-kit-react-banner@0.1.10) (2019-05-21)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.9](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.8...@workday/canvas-kit-react-banner@0.1.9) (2019-05-15)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.8](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.7...@workday/canvas-kit-react-banner@0.1.8) (2019-05-14)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.7](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.6...@workday/canvas-kit-react-banner@0.1.7) (2019-05-10)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.5...@workday/canvas-kit-react-banner@0.1.6) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.5](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.4...@workday/canvas-kit-react-banner@0.1.5) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.4](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.3...@workday/canvas-kit-react-banner@0.1.4) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.2...@workday/canvas-kit-react-banner@0.1.3) (2019-05-07)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## [0.1.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-banner/compare/@workday/canvas-kit-react-banner@0.1.1...@workday/canvas-kit-react-banner@0.1.2) (2019-05-06)

**Note:** Version bump only for package @workday/canvas-kit-react-banner





## 0.1.1 (2019-04-30)

**Note:** Version bump only for package @workday/canvas-kit-react-banner
