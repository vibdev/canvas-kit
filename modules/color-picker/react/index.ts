import ColorInput from './lib/ColorInput';
import ColorPreview from './lib/ColorPreview';

export default ColorInput;
export {ColorInput, ColorPreview};
export * from './lib/ColorInput';
export * from './lib/ColorPreview';
