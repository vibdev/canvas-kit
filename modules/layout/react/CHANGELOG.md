# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.17](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.16...@workday/canvas-kit-react-layout@0.1.17) (2019-06-17)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.16](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.15...@workday/canvas-kit-react-layout@0.1.16) (2019-06-14)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.15](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.14...@workday/canvas-kit-react-layout@0.1.15) (2019-06-10)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.14](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.13...@workday/canvas-kit-react-layout@0.1.14) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.13](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.12...@workday/canvas-kit-react-layout@0.1.13) (2019-06-05)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.12](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.11...@workday/canvas-kit-react-layout@0.1.12) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.11](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.10...@workday/canvas-kit-react-layout@0.1.11) (2019-06-03)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.10](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.9...@workday/canvas-kit-react-layout@0.1.10) (2019-05-31)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.9](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.8...@workday/canvas-kit-react-layout@0.1.9) (2019-05-24)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.8](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.7...@workday/canvas-kit-react-layout@0.1.8) (2019-05-22)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.7](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.6...@workday/canvas-kit-react-layout@0.1.7) (2019-05-21)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.6](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.5...@workday/canvas-kit-react-layout@0.1.6) (2019-05-15)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.5](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.4...@workday/canvas-kit-react-layout@0.1.5) (2019-05-14)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.4](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.3...@workday/canvas-kit-react-layout@0.1.4) (2019-05-10)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.2...@workday/canvas-kit-react-layout@0.1.3) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.2](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.1...@workday/canvas-kit-react-layout@0.1.2) (2019-05-09)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





## [0.1.1](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/compare/@workday/canvas-kit-react-layout@0.1.0...@workday/canvas-kit-react-layout@0.1.1) (2019-05-07)

**Note:** Version bump only for package @workday/canvas-kit-react-layout





# 0.1.0 (2019-04-10)


### Features

* **layout:** added fixed width prop for columns ([fe4a860](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/fe4a860))
* **layouts:** added canvas spacing to layouts ([607ed59](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/607ed59))
* **layouts:** added spacing feature ([f388d0d](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/f388d0d))
* **layouts:** content -> capWidth ([93efa87](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/93efa87))
* **layouts:** fixed type errors ([463447a](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/463447a))
* **layouts:** initial commit for layouts component ([ae29eae](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/ae29eae))
* **layouts:** limited columns to 1 through 12 ([3cbacfb](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/3cbacfb))
* **layouts:** made column children optional ([26831f3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/26831f3))
* **layouts:** removed padding from first and last element on column ([9a3e9d3](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/9a3e9d3))
* **layouts:** updated readme ([e10218d](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/e10218d))
* **layouts:** updated readme ([8fdc4ce](https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-layout/commits/8fdc4ce))
