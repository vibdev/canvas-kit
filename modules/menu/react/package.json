{
  "name": "@workday/canvas-kit-react-menu",
  "version": "0.0.24",
  "description": "A container for navigation or action items",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://ghe.megaleo.com/design/canvas-kit-react/tree/master/modules/canvas-kit-react-menu"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts"
  ],
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "clean": "rimraf dist && rimraf .build-info && mkdirp dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build": "npm-run-all --parallel build:cjs build:es6"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "menu"
  ],
  "devDependencies": {
    "@material-ui/core": "^3.9.2",
    "@types/react-transition-group": "^2.0.14",
    "@workday/canvas-system-icons-web": "^1.0.1",
    "react-transition-group": "^2.5.0"
  },
  "dependencies": {
    "@types/uuid": "^3.4.4",
    "@workday/canvas-kit-react-card": "^0.3.23",
    "@workday/canvas-kit-react-common": "^0.7.5",
    "@workday/canvas-kit-react-core": "^1.10.6",
    "@workday/canvas-kit-react-icon": "^1.7.7",
    "@workday/design-assets-types": "^0.2.3",
    "emotion": "^9.2.12",
    "react-emotion": "^9.2.12",
    "uuid": "^3.3.2"
  },
  "peerDependencies": {
    "react": ">= 16.3 < 17"
  }
}
