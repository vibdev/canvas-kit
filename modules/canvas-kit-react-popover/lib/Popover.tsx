import * as React from 'react';
import Card from '@workday/canvas-kit-react-card';
import styled from 'react-emotion';
import {SystemIcon} from '@workday/canvas-kit-react-icon';
import {
  spacing,
  CanvasDepthValue,
  CanvasSpacingValue,
  colors,
} from '@workday/canvas-kit-react-core';
import {xIcon} from '@workday/canvas-system-icons-web';

export enum PopoverPadding {
  'zero',
  's',
  'l',
}

export interface PopoverProps {
  paddingSize: PopoverPadding;
  handleClose?: () => void;
  width?: number | string;
  heading?: React.ReactNode;
  depth?: CanvasDepthValue;
}

const closeIconSpacing = 8;

const Container = styled('div')({
  position: 'relative',
  backgroundColor: colors.frenchVanilla100,
});
const CloseIconContainer = styled('div')({
  position: 'absolute',
  right: closeIconSpacing,
  cursor: 'pointer',
  top: closeIconSpacing,
});

export default class Popover extends React.Component<PopoverProps> {
  public render() {
    const {handleClose, paddingSize, width, heading, depth, ...otherProps} = this.props;
    const popoverPadding = this.definePopoverPadding(paddingSize);
    return (
      <Container {...otherProps}>
        {handleClose ? (
          <CloseIconContainer>
            <SystemIcon onClick={handleClose} icon={xIcon} />
          </CloseIconContainer>
        ) : null}
        <Card depth={depth} heading={heading} width={width} padding={popoverPadding}>
          {this.props.children}
        </Card>
      </Container>
    );
  }

  private definePopoverPadding = (paddingSize: PopoverPadding) => {
    let popOverPadding: CanvasSpacingValue = spacing.zero;
    if (paddingSize === PopoverPadding.s) {
      popOverPadding = spacing.s;
    } else if (paddingSize === PopoverPadding.l) {
      popOverPadding = spacing.l;
    } else if (paddingSize === PopoverPadding.zero) {
      popOverPadding = spacing.zero;
    }
    return popOverPadding;
  };
}
