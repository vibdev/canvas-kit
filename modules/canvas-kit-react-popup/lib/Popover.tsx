import * as React from 'react';
import Card from '@workday/canvas-kit-react-card';
import styled from 'react-emotion';
import {IconButton} from '@workday/canvas-kit-react-button';
import {CanvasDepthValue, colors} from '@workday/canvas-kit-react-core';
import {xIcon} from '@workday/canvas-system-icons-web';

export enum PopoverPadding {
  zero = '0px',
  s = '16px',
  l = '32px',
}

export interface PopoverProps {
  padding?: PopoverPadding;
  handleClose?: () => void;
  width?: number | string;
  heading?: React.ReactNode;
  depth?: CanvasDepthValue;
}

const closeIconSpacing = 12;

const Container = styled('div')({
  position: 'relative',
  backgroundColor: colors.frenchVanilla100,
});

const CloseIconContainer = styled('div')({
  position: 'absolute',
  right: closeIconSpacing,
  cursor: 'pointer',
  top: closeIconSpacing,
});

export default class Popover extends React.Component<PopoverProps> {
  static padding = PopoverPadding;
  static defaultProps = {
    padding: Popover.padding.l,
  };
  public render() {
    const {handleClose, padding, width, heading, depth, ...otherProps} = this.props;
    return (
      <Container {...otherProps}>
        {handleClose && (
          <CloseIconContainer>
            <IconButton buttonType={IconButton.Types.Plain} onClick={handleClose} icon={xIcon} />
          </CloseIconContainer>
        )}
        <Card depth={depth} heading={heading} width={width} padding={padding}>
          {this.props.children}
        </Card>
      </Container>
    );
  }
}
