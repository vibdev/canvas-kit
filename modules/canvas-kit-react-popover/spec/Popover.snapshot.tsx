import * as React from 'react';
import Popover, {PopoverPadding} from '../lib/Popover';
import * as renderer from 'react-test-renderer';

describe('Toggle Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(<Popover paddingSize={PopoverPadding.s} />);
    expect(component).toMatchSnapshot();
  });
  test('renders popover with close icon', () => {
    const component = renderer.create(
      <Popover handleClose={jest.fn()} paddingSize={PopoverPadding.s} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders popover different padding', () => {
    const component = renderer.create(
      <Popover handleClose={jest.fn()} paddingSize={PopoverPadding.l} />
    );
    expect(component).toMatchSnapshot();
  });
  test('renders popover with children elements', () => {
    const component = renderer.create(
      <Popover handleClose={jest.fn()} paddingSize={PopoverPadding.zero}>
        <span>hello world</span>
      </Popover>
    );
    expect(component).toMatchSnapshot();
  });
  test('renders without close icon', () => {
    const component = renderer.create(
      <Popover paddingSize={PopoverPadding.zero}>
        <span>hello world</span>
      </Popover>
    );
    expect(component).toMatchSnapshot();
  });
});
