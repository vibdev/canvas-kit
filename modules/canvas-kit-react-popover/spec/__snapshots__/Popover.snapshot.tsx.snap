// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Snapshots renders as expected 1`] = `
.emotion-4 {
  position: relative;
  background-color: #ffffff;
}

.emotion-2 {
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 16px;
}

.emotion-0 {
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

<div
  className="emotion-4 emotion-5"
>
  <div
    className="emotion-2 emotion-3"
  >
    <div
      className="emotion-0 emotion-1"
    />
  </div>
</div>
`;

exports[`Toggle Snapshots renders popover different padding 1`] = `
.emotion-7 {
  position: relative;
  background-color: #ffffff;
}

.emotion-3 {
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-1 {
  position: absolute;
  right: 8px;
  cursor: pointer;
  top: 8px;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-5 {
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 32px;
}

<div
  className="emotion-7 emotion-8"
>
  <div
    className="emotion-1 emotion-2"
  >
    <span
      className="emotion-0"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
      onClick={[MockFunction]}
    />
  </div>
  <div
    className="emotion-5 emotion-6"
  >
    <div
      className="emotion-3 emotion-4"
    />
  </div>
</div>
`;

exports[`Toggle Snapshots renders popover with children elements 1`] = `
.emotion-7 {
  position: relative;
  background-color: #ffffff;
}

.emotion-3 {
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-1 {
  position: absolute;
  right: 8px;
  cursor: pointer;
  top: 8px;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

.emotion-5 {
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 0px;
}

<div
  className="emotion-7 emotion-8"
>
  <div
    className="emotion-1 emotion-2"
  >
    <span
      className="emotion-0"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
      onClick={[MockFunction]}
    />
  </div>
  <div
    className="emotion-5 emotion-6"
  >
    <div
      className="emotion-3 emotion-4"
    >
      <span>
        hello world
      </span>
    </div>
  </div>
</div>
`;

exports[`Toggle Snapshots renders popover with close icon 1`] = `
.emotion-7 {
  position: relative;
  background-color: #ffffff;
}

.emotion-5 {
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 16px;
}

.emotion-3 {
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-1 {
  position: absolute;
  right: 8px;
  cursor: pointer;
  top: 8px;
}

.emotion-0 {
  display: inline-block;
}

.emotion-0 svg {
  display: block;
}

.emotion-0 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-0 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-0:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-0 .wd-icon-background {
  fill: transparent;
}

.emotion-0:hover .wd-icon-background {
  fill: transparent;
}

<div
  className="emotion-7 emotion-8"
>
  <div
    className="emotion-1 emotion-2"
  >
    <span
      className="emotion-0"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-x wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M5.996 4.582a.5.5 0 0 0-.717-.003l-.7.7a.504.504 0 0 0 .003.717L10.586 12l-6.004 6.004a.5.5 0 0 0-.003.717l.7.7c.197.197.514.2.717-.003L12 13.414l6.004 6.004a.5.5 0 0 0 .717.003l.7-.7a.504.504 0 0 0-.003-.717L13.414 12l6.004-6.004a.5.5 0 0 0 .003-.717l-.7-.7a.504.504 0 0 0-.717.003L12 10.586 5.996 4.582z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
      onClick={[MockFunction]}
    />
  </div>
  <div
    className="emotion-5 emotion-6"
  >
    <div
      className="emotion-3 emotion-4"
    />
  </div>
</div>
`;

exports[`Toggle Snapshots renders without close icon 1`] = `
.emotion-4 {
  position: relative;
  background-color: #ffffff;
}

.emotion-0 {
  font-size: 14px;
  color: #494949;
  font-weight: 400;
  line-height: 20px;
  font-family: "Roboto","Helvetica Neue","Helvetica",Arial,sans-serif;
}

.emotion-2 {
  border: 1px solid #ced3d9;
  border-radius: 3px;
  box-sizing: border-box;
  box-shadow: 0px 4px 8px 0 rgba(0,0,0,0.1);
  padding: 0px;
}

<div
  className="emotion-4 emotion-5"
>
  <div
    className="emotion-2 emotion-3"
  >
    <div
      className="emotion-0 emotion-1"
    >
      <span>
        hello world
      </span>
    </div>
  </div>
</div>
`;
